test/connectivity/modules/NetworkController/Program.cs

```diff
@@ -17,37 +17,41 @@ static async Task Main()
         {
             (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Log);
 
-            Log.LogInformation($"Starting with {Settings.Current.NetworkControllerMode}");
+            Log.LogInformation($"Starting with {Settings.Current.NetworkRunProfile.ProfileType}");
 
             try
             {
                 var networkInterfaceName = DockerHelper.GetDockerInterfaceName();
 
-                if (networkInterfaceName.HasValue)
-                {
-                    await networkInterfaceName.ForEachAsync(
-                        async name =>
+                await networkInterfaceName.ForEachAsync(
+                    async name =>
+                    {
+                        var offline = new OfflineController(name, Settings.Current.IotHubHostname, Settings.Current.NetworkRunProfile.ProfileSettings);
+                        var satellite = new SatelliteController(name, Settings.Current.IotHubHostname, Settings.Current.NetworkRunProfile.ProfileSettings);
+                        var cellular = new CellularController(name, Settings.Current.IotHubHostname, Settings.Current.NetworkRunProfile.ProfileSettings);
+                        var controllers = new List<INetworkController>() { offline, satellite, cellular };
+                        await RemoveAllControllingRules(controllers, cts.Token);
+
+                        Enum.TryParse(Settings.Current.NetworkRunProfile.ProfileType, out NetworkControllerType ncType);
+                        switch (ncType)
                         {
-                            var firewall = new FirewallOfflineController(name, Settings.Current.IotHubHostname);
-                            var satellite = new SatelliteController(name);
-                            var controllers = new List<INetworkController>() { firewall, satellite };
-                            await RemoveAllControllingRules(controllers, cts.Token);
-
-                            switch (Settings.Current.NetworkControllerMode)
-                            {
-                                case NetworkControllerMode.OfflineTrafficController:
-                                    await StartAsync(firewall, cts.Token);
-                                    break;
-                                case NetworkControllerMode.SatelliteTrafficController:
-                                    await StartAsync(satellite, cts.Token);
-                                    break;
-                            }
-                        });
-                }
-                else
-                {
-                    Log.LogError($"No network interface found for docker network {Settings.Current.NetworkId}");
-                }
+                            case NetworkControllerType.Offline:
+                                await StartAsync(offline, cts.Token);
```

Comment:
why not create offline controller inside this switch case?

Reply:
because it creates all controllers to remove all rules in the beginning